internal/pkg/describe/env.go

```diff
@@ -146,6 +150,28 @@ func (d *EnvDescriber) Version() (string, error) {
 	return metadata.Version, nil
 }
 
+// ServiceDiscoveryEndpoint returns the endpoint the environment was initialized with, if any. Otherwise,
+// it returns the legacy app.local endpoint.
+func (d *EnvDescriber) ServiceDiscoveryEndpoint() (string, error) {
+	p, err := d.Params()
+	if err != nil {
+		return "", fmt.Errorf("get params of environment %s in app %s: %w", d.env.Name, d.env.App, err)
+	}
+	for k, v := range p {
+		// Ignore non-svc discovery params
+		if k != cfnstack.EnvParamServiceDiscoveryEndpoint {
+			continue
+		}
+		// If the service discovery param is set (meaning the stack is not upgraded), use the provided value
+		if v != "" {
```

Comment:
Wouldn't this mean that the stack is actually upgraded? i.e. we are at least on v1.5.0